src/main/java/org/skriptlang/skript/lang/experiment/ExperimentData.java

```diff
@@ -0,0 +1,141 @@
+package org.skriptlang.skript.lang.experiment;
+
+import ch.njol.skript.Skript;
+import ch.njol.util.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+/**
+ * Container for holding {@link Experiment}s that must be enabled or disabled to use.
+ */
+public class ExperimentData {
+
+	private Experiment @Nullable [] required = null;
+	private Experiment @Nullable [] disallowed = null;
+	private @Nullable String errorMessage = null;
+
+	public ExperimentData() {}
+
+	/**
+	 * Set the {@link Experiment}s that must be enabled in order to use.
+	 * @param required The {@link Experiment}s
+	 * @return This {@link ExperimentData}.
+	 */
+	public ExperimentData required(Experiment... required) {
+		this.required = required;
+		return this;
+	}
+
+	/**
+	 * Set the {@link Experiment}s that must be disabled in order to use.
+	 * @param disallowed The {@link Experiment}s
+	 * @return This {@link ExperimentData}.
+	 */
+	public ExperimentData disallowed(Experiment... disallowed) {
+		this.disallowed = disallowed;
+		return this;
+	}
+
+	/**
+	 * Set the error message to be printed if the requirements of this {@link ExperimentData} are not met.
+	 * @param errorMessage The error message.
+	 * @return This {@link ExperimentData}.
+	 */
+	public ExperimentData errorMessage(String errorMessage) {
+		this.errorMessage = errorMessage;
+		return this;
+	}
+
+	/**
+	 * Get the {@link Experiment}s that must be enabled in order to use.
+	 */
+	public Experiment @Nullable [] getRequired() {
+		return required;
+	}
+
+	/**
+	 * Get the {@link Experiment}s that must be disabled in order to use.
+	 */
+	public Experiment @Nullable [] getDisallowed() {
+		return disallowed;
+	}
+
+	/**
+	 * Check if this {@link ExperimentData} is valid.
+	 * @return {@code True} if valid.
+	 */
+	public boolean isValid() {
+		return required != null || disallowed != null;
+	}
+
+	/**
+	 * Check if the requirements of this {@link ExperimentData} are met.
+	 * @param experiments The current enabled {@link Experiment}s.
+	 * @return {@code True} if the requirements were met.
+	 */
```

Comment:
should note that it prints errors in here
It may be worth having the actual checking logic be one method and the error printing be a different method that calls the first.